### ISSUE TITLE
The page suddenly crashes and all UI disappears

### ISSUE THREAD

**Issue by wellmoonloft** [afadil/wealthfolio#157]
version v1.0.19 and v1.0.20
It happens occasionally when opening the app, and also after a portfolio update.
Since I can't see the error message, I guess the problem is caused by a network error when getting the latest market data.
I think can make a judgment that when getting market data fails, it can use the old data and prompt the user to refresh manually instead of letting the page crash directly.
Of course, this is my guess.

<img width="1357" alt="image" src="https://github.com/user-attachments/assets/32b6c961-a4c3-49b1-ab84-c2f7ccca4c27">

**Comment by wellmoonloft** [comment]
<img width="1665" alt="image" src="https://github.com/user-attachments/assets/429146a6-3d0a-41ca-8c8d-d94d06c5d3d8">
This situation is very frequent now, it happens almost every time
When I enter or modify an activity in Activities, and then return to Dashboard, the UI will be stuck
The last log is `Fetching active accounts performance...`
I hope to be able to track the exception and find the cause, but I really don't understand Tauri and Rust

**Comment by afadil** [comment]
Seems like a frontend Error. Can you please Inspect the page and see if there is any errors in the console.
I'm working on a new update with better logs using tauri-plugin-log. 
To inspect run the app in dev mode then right click and Inspect Element
![Screenshot 2024-11-01 at 3 59 03 PM](https://github.com/user-attachments/assets/59d976ed-e0a1-4d3d-a102-f850d4f924ac)

**Comment by wellmoonloft** [comment]
Thanks for your reply. I just tested it and it now occurs 100% of the time.
1. Open the app
2. Enter Activities
3. Add or modify a piece of data
4. Return to Dashboard
Error occurs
<img width="1354" alt="image" src="https://github.com/user-attachments/assets/7a854707-8826-4b4f-adfd-015a2ceea76e">

```
[Error] TypeError: Type error
	(anonymous function) (chunk-DVCT6SLB.js:16670)
	callCallback2 (chunk-DVCT6SLB.js:3674)
	dispatchEvent
	invokeGuardedCallbackDev (chunk-DVCT6SLB.js:3699)
	invokeGuardedCallback (chunk-DVCT6SLB.js:3733)
	reportUncaughtErrorInDEV (chunk-DVCT6SLB.js:16669)
	captureCommitPhaseError (chunk-DVCT6SLB.js:19558)
	commitLayoutMountEffects_complete (chunk-DVCT6SLB.js:17982)
	commitLayoutEffects_begin (chunk-DVCT6SLB.js:17969)
	commitLayoutEffects (chunk-DVCT6SLB.js:17920)
	commitRootImpl (chunk-DVCT6SLB.js:19353)
	commitRoot (chunk-DVCT6SLB.js:19277)
	performSyncWorkOnRoot (chunk-DVCT6SLB.js:18895)
	performSyncWorkOnRoot
	flushSyncCallbacks (chunk-DVCT6SLB.js:9119)
	(anonymous function) (chunk-DVCT6SLB.js:18627)
[Error] The above error occurred in the <NumberFlowImpl> component:

NumberFlowImpl@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:485:10
NumberFlow2@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:560:63
h1
Balance@http://localhost:1420/src/pages/dashboard/balance.tsx:24:32
div
a
Primitive.a@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
SlotClone@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:53:13
Slot@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:34:31
Primitive.div@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
PopperAnchor@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1914:38
HoverCardTrigger@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:111:40
PopperProvider@http://localhost:1420/node_modules/.vite/deps/chunk-GVVDIM6W.js:21:42
Popper@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1907:37
HoverCardProvider@http://localhost:1420/node_modules/.vite/deps/chunk-D2PQHPUZ.js:38:42
HoverCard@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:61:14
div
div
DashboardPage@http://localhost:1420/src/pages/dashboard/dashboard-page.tsx:77:43
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Outlet@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4467:25
div
main
div
div
AppLayout@http://localhost:1420/src/pages/layouts/app-layout.tsx:78:71
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Routes@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4534:19
Router@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4477:28
BrowserRouter@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:5224:26
AppRoutes@http://localhost:1420/src/routes.tsx:38:25
SettingsProvider@http://localhost:1420/src/lib/settings-provider.tsx:24:5
QueryClientProvider@http://localhost:1420/node_modules/.vite/deps/@tanstack_react-query.js:2791:9
App@http://localhost:1420/src/App.tsx:24:33

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
	logCapturedError (chunk-DVCT6SLB.js:14032)
	(anonymous function) (chunk-DVCT6SLB.js:14052)
	callCallback (chunk-DVCT6SLB.js:11248)
	commitUpdateQueue (chunk-DVCT6SLB.js:11265)
	commitLayoutEffectOnFiber (chunk-DVCT6SLB.js:17093)
	commitLayoutMountEffects_complete (chunk-DVCT6SLB.js:17980)
	commitLayoutEffects_begin (chunk-DVCT6SLB.js:17969)
	commitLayoutEffects (chunk-DVCT6SLB.js:17920)
	commitRootImpl (chunk-DVCT6SLB.js:19353)
	commitRoot (chunk-DVCT6SLB.js:19277)
	performSyncWorkOnRoot (chunk-DVCT6SLB.js:18895)
	performSyncWorkOnRoot
	flushSyncCallbacks (chunk-DVCT6SLB.js:9119)
	(anonymous function) (chunk-DVCT6SLB.js:18627)
[Error] The above error occurred in the <NumberFlowImpl> component:

NumberFlowImpl@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:485:10
NumberFlow2@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:560:63
div
div
GainAmount@http://localhost:1420/src/components/gain-amount.tsx:27:32
div
div
a
Primitive.a@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
SlotClone@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:53:13
Slot@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:34:31
Primitive.div@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
PopperAnchor@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1914:38
HoverCardTrigger@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:111:40
PopperProvider@http://localhost:1420/node_modules/.vite/deps/chunk-GVVDIM6W.js:21:42
Popper@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1907:37
HoverCardProvider@http://localhost:1420/node_modules/.vite/deps/chunk-D2PQHPUZ.js:38:42
HoverCard@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:61:14
div
div
DashboardPage@http://localhost:1420/src/pages/dashboard/dashboard-page.tsx:77:43
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Outlet@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4467:25
div
main
div
div
AppLayout@http://localhost:1420/src/pages/layouts/app-layout.tsx:78:71
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Routes@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4534:19
Router@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4477:28
BrowserRouter@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:5224:26
AppRoutes@http://localhost:1420/src/routes.tsx:38:25
SettingsProvider@http://localhost:1420/src/lib/settings-provider.tsx:24:5
QueryClientProvider@http://localhost:1420/node_modules/.vite/deps/@tanstack_react-query.js:2791:9
App@http://localhost:1420/src/App.tsx:24:33

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
	logCapturedError (chunk-DVCT6SLB.js:14032)
	(anonymous function) (chunk-DVCT6SLB.js:14052)
	callCallback (chunk-DVCT6SLB.js:11248)
	commitUpdateQueue (chunk-DVCT6SLB.js:11265)
	commitLayoutEffectOnFiber (chunk-DVCT6SLB.js:17093)
	commitLayoutMountEffects_complete (chunk-DVCT6SLB.js:17980)
	commitLayoutEffects_begin (chunk-DVCT6SLB.js:17969)
	commitLayoutEffects (chunk-DVCT6SLB.js:17920)
	commitRootImpl (chunk-DVCT6SLB.js:19353)
	commitRoot (chunk-DVCT6SLB.js:19277)
	performSyncWorkOnRoot (chunk-DVCT6SLB.js:18895)
	performSyncWorkOnRoot
	flushSyncCallbacks (chunk-DVCT6SLB.js:9119)
	(anonymous function) (chunk-DVCT6SLB.js:18627)
[Error] The above error occurred in the <NumberFlowImpl> component:

NumberFlowImpl@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:485:10
NumberFlow2@http://localhost:1420/node_modules/.vite/deps/@number-flow_react.js:560:63
div
div
GainPercent@http://localhost:1420/src/components/gain-percent.tsx:21:32
div
div
a
Primitive.a@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
SlotClone@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:53:13
Slot@http://localhost:1420/node_modules/.vite/deps/chunk-QFXECBFQ.js:34:31
Primitive.div@http://localhost:1420/node_modules/.vite/deps/chunk-QYPLS2RT.js:42:32
PopperAnchor@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1914:38
HoverCardTrigger@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:111:40
PopperProvider@http://localhost:1420/node_modules/.vite/deps/chunk-GVVDIM6W.js:21:42
Popper@http://localhost:1420/node_modules/.vite/deps/chunk-VOOFSE4D.js:1907:37
HoverCardProvider@http://localhost:1420/node_modules/.vite/deps/chunk-D2PQHPUZ.js:38:42
HoverCard@http://localhost:1420/node_modules/.vite/deps/@radix-ui_react-hover-card.js:61:14
div
div
DashboardPage@http://localhost:1420/src/pages/dashboard/dashboard-page.tsx:77:43
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Outlet@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4467:25
div
main
div
div
AppLayout@http://localhost:1420/src/pages/layouts/app-layout.tsx:78:71
RenderedRoute@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4096:32
Routes@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4534:19
Router@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:4477:28
BrowserRouter@http://localhost:1420/node_modules/.vite/deps/react-router-dom.js:5224:26
AppRoutes@http://localhost:1420/src/routes.tsx:38:25
SettingsProvider@http://localhost:1420/src/lib/settings-provider.tsx:24:5
QueryClientProvider@http://localhost:1420/node_modules/.vite/deps/@tanstack_react-query.js:2791:9
App@http://localhost:1420/src/App.tsx:24:33

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
	logCapturedError (chunk-DVCT6SLB.js:14032)
	(anonymous function) (chunk-DVCT6SLB.js:14052)
	callCallback (chunk-DVCT6SLB.js:11248)
	commitUpdateQueue (chunk-DVCT6SLB.js:11265)
	commitLayoutEffectOnFiber (chunk-DVCT6SLB.js:17093)
	commitLayoutMountEffects_complete (chunk-DVCT6SLB.js:17980)
	commitLayoutEffects_begin (chunk-DVCT6SLB.js:17969)
	commitLayoutEffects (chunk-DVCT6SLB.js:17920)
	commitRootImpl (chunk-DVCT6SLB.js:19353)
	commitRoot (chunk-DVCT6SLB.js:19277)
	performSyncWorkOnRoot (chunk-DVCT6SLB.js:18895)
	performSyncWorkOnRoot
	flushSyncCallbacks (chunk-DVCT6SLB.js:9119)
	(anonymous function) (chunk-DVCT6SLB.js:18627)
[Error] TypeError: Type error
	flushSyncCallbacks (chunk-DVCT6SLB.js:9129)
	(anonymous function) (chunk-DVCT6SLB.js:18627)
> Selected Element
< <html lang="en" style="color-scheme: light;">
<head>
<script type="module">…</script>
<script type="module" src="/@vite/client"></script>
<meta charset="UTF-8">
<link rel="icon" type="image/svg+xml" href="/vite.svg">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Wealthfolio</title>
<style type="text/css" data-vite-dev-id="/Users/linan/wealthfolio/src/styles.css">…</style>
</head>
<body class="light font-mono" style>
<div id="root"></div>
<script type="module" src="/src/main.tsx"></script>
</body>
</html>
```

**Comment by wellmoonloft** [comment]
it fix on v1.0.21
